### ISSUE TITLE
[Feature request] Animated PNG (APNG) support

### ISSUE THREAD

**Issue by Rongronggg9** [AaronFeng753/Waifu2x-Extension-GUI#239]
> In a comparison made between GIF, APNG and WebP, it was shown that APNG kept lower file size while keeping at least equal quality.
[Wikipedia](https://en.wikipedia.org/wiki/APNG)

Some animated stickers are distributed in APNG format instead of GIF (e.g. Line sticker), but this GUI could only recognize and process them as normal PNGs. I've noticed that both ffmpeg and ImageMagick (>=7.0.10-31) support APNG, which means that pre-processing and post-processing APNG in the similar way to GIF may be feasible.

**Comment by AaronFeng753** [comment]
> ImageMagick (>=7.0.10-31) support APNG

ImageMagick barely supports APNG, but I found other tools by this guy: https://sourceforge.net/u/maxst/profile/

**Comment by AaronFeng753** [comment]
APNG support will be available in the next update
✅ https://github.com/AaronFeng753/Waifu2x-Extension-GUI/commit/43eb050382ef950be585d0698bb29e5b5ce4dbbc

**Comment by Rongronggg9** [comment]
> 
> 
> APNG support will be available in the next update
> ✅ [43eb050](https://github.com/AaronFeng753/Waifu2x-Extension-GUI/commit/43eb050382ef950be585d0698bb29e5b5ce4dbbc)

That's awesome!

**Comment by AaronFeng753** [comment]
Now it's available in the latest BETA version 🎉

Support this project on patreon:
https://www.patreon.com/aaronfeng

**Comment by Rongronggg9** [comment]
**It may wrongly insist that there are twice as many frames than what the APNG actually has.**

For example, [377520094@2x.png](https://user-images.githubusercontent.com/15956627/116280581-6929c480-a7bb-11eb-9337-e259a319ca1e.png) is an APNG with 20 frames and will be splitted to 20 PNGs after starting processing (that's right). However, W2xEX will insist that the APNG has 40 frames. Once the first 20 PNGs are poccessed, it will try to process the last 20 PNGs, which do not exist, only to fail. 
Renaming the extension to apng will not make any difference. 

![Screenshot](https://user-images.githubusercontent.com/15956627/116282394-47c9d800-a7bd-11eb-8458-034eec783490.png)

**Comment by AaronFeng753** [comment]
> It may wrongly insist that there are twice as many frames than what the APNG actually has.

I download that image but I can't recreate this bug, this might be caused by GUI failed to delete `.txt` files generated by the disassembler. Could you check the `splitFrames` folder?

**Comment by AaronFeng753** [comment]
I tried several times but I still can't recreate this bug, but I found another one: https://github.com/AaronFeng753/Waifu2x-Extension-GUI/commit/e760d108e12f891816909727f70438e9b99b2252

**Comment by Rongronggg9** [comment]
Hmm... I did more tests and found it only occurs when choosing waifu2x-ncnn-vulkan as the engine. Path or detailed settings seem to be irrelevant.

> this might be caused by GUI failed to delete .txt files generated by the disassembler. Could you check the `splitFrames` folder?

No .txt in the `splitFrames` folder.

> Could you check the `splitFrames` folder?

![Screenshot_1](https://user-images.githubusercontent.com/15956627/116351630-8fd11500-a826-11eb-99fc-2c0ea6e1eab8.png)

> And send me the full path of the `splitFrames` folder.

I've tried to place the image to different folders only to get the same result. Here's the full paths to `splitFrames` folders:
```
C:\Users\Rongrong\OneDrive\Sticker\Capoo\貓貓蟲-咖波 好運發大財 賀年貼圖\animation@2x\377520094@2x_png_splitFramesFolder_W2xEX
C:\Users\Rongrong\OneDrive\Sticker\Capoo\2x\animated\377520094@2x_png_splitFramesFolder_W2xEX
C:\Users\Rongrong\377520094@2x_png_splitFramesFolder_W2xEX
```

**Comment by AaronFeng753** [comment]
> only occurs when choosing waifu2x-ncnn-vulkan as the engine

Okay, now I know what happened, you are using `waifu2x-ncnn-vulkan` engine and the `Scale Ratio is 4x`, so the engine need to process those frames `2 times` to make it `4x` larger, that's why current file progress shows `20/40`, which means it just finished the first round(make the images 2x larger)

And the text box at the bottom is keep saying `automatic retry`, that means the engine is crashing and the GUI is trying to start it again.

In conclusion, you `don't have enough VRAM or RAM`, so the engine can't enlarge those images to `4x`, there is no bug in the GUI, I just tested it and everything works just fine on my PC.

Just try another engine or change the engines settings.

**Comment by AaronFeng753** [comment]
Please create new issues if you found another bug or have other suggestions
